src/Type/Definition/FieldArgument.php

```diff
@@ -66,12 +67,21 @@ public function __construct(array $def)
     }
 
     /**
-     * @param mixed[] $config
+     * @param mixed[]|callable $config
      *
      * @return FieldArgument[]
+     * @throws \InvalidArgumentException
      */
-    public static function createMap(array $config)
+    public static function createMap($config)
     {
+        if (is_callable($config)) {
```

Comment:
Sorry for the long delay, but as promised I tried out your changes. So far I don't quite understand how this changes things; `createMap` is called by the `FieldDefinition` constructor, and then you just check to see if `$config` is callable, and if it is you immediately call it? Where do the savings come in? 

I tested this by converting a few of the `args` on my top-level `Query` fields to functions, and stuck breakpoints inside them; they get called immediately.

What am I missing, here?

Reply:
I just POC'ed my own full lazy loading concept; so far seems to work pretty well. I'll try to put together a PR by the end of the week.